### ISSUE TITLE
CharonSession() can time out I think

### ISSUE THREAD

**Issue by mariogiov** [NationalGenomicsInfrastructure/ngi_pipeline#207]
Technically I think the CharonSession http session can time out now that this class is implemented as a singleton. This should not happen in normal cases but it should be handled in the code somewhere.

**Comment by mariogiov** [comment]
I was wrongo bongo about this, each call to Charon opens a new connection.

**Comment by Galithil** [comment]
Expanding a little bit on this :
Charon session does not hold a connection open, because it only contacts API endpoints.
These API endpoints give out JSON data, and then close the connection, so you have to reopen one every time. This is closer to REST than to an actual DB connection

**Comment by guillermo-carrasco** [comment]
and wasn't that the idea of Charon @Galithil ? To have a REST interface over CouchDB? (which is by the way another REST interface :sweat_smile: )

**Comment by Galithil** [comment]
pretty much. I'm not saying that's a good thing, but at least, we don't have to worry over connections

**Comment by pekrau** [comment]
I'm not sure I understand the issue, but the Charon JSON interface is RESTful, and as such is stateless, and sessions do not enter into it. And the idea of using Charon REST, rather than CouchDB REST is just to add domain logic to the interface, and a layer doing data joins (for reading) and input checking (for writing). The Python requests package (which I use instead of the standard urllib2) does have a Session, which possibly does what you want. Anyway, why do you want to keep a connection open?

**Comment by mariogiov** [comment]
This thread has garnered after I closed it more comments than any of the open issues. I don't want to keep a connection open, I just misunderstood the way requests.Session worked: I thought it kept an http connection open, and I was concerned that after some period it would time out and the CharonSession object would no longer be able to communicate with Charon. This proves not to be an issue, and so I have closed it. RIP this issue
